File: app/src/main/java/com/gmail/laktionov/sample/rx/lyricsfinder/version1/search/datasource/remote/RemoteSource.java
package com.gmail.laktionov.sample.rx.lyricsfinder.version1.search.datasource.remote;

import com.gmail.laktionov.sample.rx.lyricsfinder.version1.search.datasource.remote.model.SearchResponse;

import io.reactivex.Maybe;

public interface RemoteSource {
    Maybe<SearchResponse> searchLyric(String artistName, String artistSong);
}

File: app/src/main/java/com/gmail/laktionov/lyricsfinder/domain/datasource/RemoteSource.kt
package com.gmail.laktionov.lyricsfinder.domain.datasource

import com.gmail.laktionov.lyricsfinder.domain.BaseResponse
import com.gmail.laktionov.lyricsfinder.domain.SongLyric

interface RemoteSource {

    fun findLyricRemote(artistName: String, songName: String): BaseResponse<SongLyric>
}